src/jdk.jshell/share/classes/jdk/jshell/execution/LocalExecutionControl.java

```diff
@@ -40,6 +48,7 @@ public class LocalExecutionControl extends DirectExecutionControl {
     private final Object STOP_LOCK = new Object();
     private boolean userCodeRunning = false;
     private ThreadGroup execThreadGroup;
+    private Field allStop = null;
```

Comment:
It might be better to use a `VarHandle` for this (as it doesn’t need boxing or access checks on every access).

Reply:
According to javadoc VarHandle of static field "effectively ignore the volatile declaration".
Volatility of REPL.$Cancel.allStop field is critical for this use case.

Reply:
> According to javadoc VarHandle of static field "effectively ignore the volatile declaration".
> Volatility of REPL.$Cancel.allStop field is critical for this use case.

VarHandles defines xxxVolatile so it would be feasible but I don't see a strong need to change it as the reflective op is only done at the start to reset "stop" and when control-C/stop is done.